File: c/RemoveDuplicatesfromSortedArray.c
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

int removeDuplicates(int* nums, int numsSize){
    
    if (numsSize <= 1)
        return numsSize;
    
    int j = 1;
    
    for (int i = 1; i < numsSize; i++)
        if (nums[i] != nums[j - 1])
            nums[j++] = nums[i];
    
    return j;
}

File: swift/RemoveDuplicatesFromSortedArray.swift
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        var k = 0
        for i in 0..<nums.count {
            if nums[k] != nums[i] {
                k += 1
                nums[k] = nums[i]
            }
        }
        return k + 1
    }
}
